### PR TITLE
`openMRS` adding more functions

### PR DESCRIPTION
**Pull request by mtuchi**
## Implementation Plan
Add the following helper functions in `OpenMRS` adaptor
- [x]  searchPatient
- [x] searchPerson
- [x] createPatient
- [x] getPatient
- [x] getEncounter
- [x] getEncounters
- [x] createEncounter

## Issues

Reference any related issues here. #255 

## Review Checklist

Before merging, the reviewer should check the following items:

- [x] Does the PR do what it claims to do?
- [ ] If this is a new adaptor, has the migration tool been run and the
      migration guide followed?
- [ ] Are there any unit tests? Should there be?
- [x] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

**Comment by mtuchi** [comment]
For testing `createEncounter()`
```
createEncounter({
  encounterDatetime: '2023-05-25T06:08:25.000+0000',
  patient: '1fdaa696-e759-4a7d-a066-f1ae557c151b',
  encounterType: 'dd528487-82a5-4082-9c72-ed246bd49591',
  location: 'ba685651-ed3b-4e63-9b35-78893060758a',
  encounterProviders: [],
  visit: {
    patient: '1fdaa696-e759-4a7d-a066-f1ae557c151b',
    visitType: '7b0f5697-27e3-40c4-8bae-f4049abfb4ed',
    startDatetime: '2023-05-25T06:08:25.000+0000',
    stopDatetime: '2023-05-25T06:09:25.000+0000',
  },
});
```

**Review by stuartc — CHANGES_REQUESTED** [review]
@mtuchi nice refactoring here, we need to address two things:

- Not wrapping `superagent` in promises since it already returns a promise.
- Making sure that those final catches aren't going to break when we throw custom errors.

File: packages/openmrs/src/Adaptor.js [hunk 1]
```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
...
-        const resp = tryJson(body);
-        resolve({ ...state, auth: resp });
-      }
-    });
+    agent
+      .get(`${instanceUrl}/ws/rest/v1/session`)
+      .auth(username, password)
+      .then(() => {
+        resolve({ ...state, agent });
+      });
```
Comment (stuartc) [inline packages/openmrs/src/Adaptor.js:54]: I'm not sure what is going on here, it doesn't need to be wrapped in a promise; and you don't seem to be getting the response from the server?

File: packages/openmrs/src/Adaptor.js [hunk 2]
```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
...
+      .end((error, response) => {
+        error = assembleError({ error, response });
+        if (error) {
+          throw error;
+        } else {
+          console.log(`Success. Found patient.`);
+
+          const data = tryJson(response.text);
+          const nextState = composeNextState(state, data);
+          return nextState;
```
Comment (stuartc) [inline packages/openmrs/src/Adaptor.js:100]: I see you're using `.end` here, and in other places. We need to pick one, either use the `.then` pattern or `.end`.

File: packages/openmrs/src/Adaptor.js [hunk 3]
```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
...
+      .then(response => {
+        console.log(`Successfull created an encounter.`);
+
+        const data = tryJson(response.text);
+        const nextState = composeNextState(state, data);
+        return nextState;
+      })
+      .catch(error => {
+        throw error;
+      });
```
Comment (stuartc) [inline packages/openmrs/src/Adaptor.js:149]: You don't need to catch to throw an error. This is redundant.

File: packages/openmrs/src/Adaptor.js [hunk 4]
```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
...
+      .then(response => {
+        const data = tryJson(response.text);
+        const nextState = composeNextState(state, data);
+
+        if (callback) return callback(nextState);
+        return nextState;
+      })
+      .catch(error => {
+        throw error;
+      });
```
Comment (stuartc) [inline packages/openmrs/src/Adaptor.js:188]: Please remove this.

File: packages/openmrs/src/Adaptor.js [hunk 5]
```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
...
-
-            if (options.exactlyOne && data.results.length !== 1) {
-              reject(`Raising an error because ${count} records were found.`);
-            } else {
-              console.log(
-                `Search successful. Returned ${count} patient${
-                  count > 1 ? 's' : ''
-                }.`
-              );
-            }
```
Comment (stuartc) [inline packages/openmrs/src/Adaptor.js:160]: What happened to these logs/errors?

**Review by stuartc — CHANGES_REQUESTED** [review]
@mtuchi I'm not stoked with these changes. How are we testing this?

I see the appeal to use `superagent` instead of `request`, but there seems to be a lot of changes that either seem to be unnecessary or ignore previous functionality.

Lets pair to get this sorted.

**Pull request merged**